### PR TITLE
prov/efa: Fix FI_HMEM backwards compat

### PR DESCRIPTION
**Pull request by rwespetal**
Two commits here, one to fix the capability bits internally so that we can look at the values in util_domain and another to fix the actual issue.

### CODE REVIEW

**Comment by rwespetal** [comment]
bot:aws:retest

File: prov/efa/src/efa_mr.c [hunk 1]
```diff
@@ -315,7 +318,14 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
...
-	efa_mr->peer.iface = mr_attr->iface;
+	/*
+	 * Skipping the domain type check is okay here since util_domain is at
+	 * the beginning of efa_domain and rxr_domain.
+	 */
+	if (efa_mr->domain->util_domain.info_domain_caps & FI_HMEM)
+		efa_mr->peer.iface = mr_attr->iface;
+	else
+		efa_mr->peer.iface = FI_HMEM_SYSTEM;
 	if (mr_attr->iface == FI_HMEM_CUDA)
```
Comment (shefty) [inline prov/efa/src/efa_mr.c:329]: I think you need to update this check to use efa_mr->peer.iface

Comment (rwespetal) [reply prov/efa/src/efa_mr.c]: Thank you, fixed.

**Comment by rwespetal** [comment]
The NCCL failure in AWS CI is real so we're looking at that.

**Comment by rwespetal** [comment]
Switched over to use the EFA GPU direct sysfs file based on some feedback. I'm still trying to track down whether the NCCL failure is real, the last CI run had a different failure than the first run.

**Comment by rwespetal** [comment]
We did some more internal testing with NCCL and are not seeing an issue. I'll file an issue internally for the NCCL CI failures we saw for this PR.

**Pull request merged**